### PR TITLE
76 autopipe

### PR DESCRIPTION
**Pull request by KlamChowder1**
tried a few different approaches but ultimately went back to the way i did it before... still not sure if this is the best way to do it though :/ 

also was thinking of putting it in the pipeline builder page but not sure if that would clutter it up too much with too many buttons - so felt like having it in users settings is a nice central place to put it - but idk if maybe it should be in the DicomRuns page or elsewhere perhaps

autopipe
image

### CODE REVIEW

**Comment by KlamChowder1** [comment]
stores in db too now 

autopipe_crud

**Comment by awtkns** [comment]
@KlamChowder1 so what's missing now, just the actual pipelining integration?

File: backend/api/routes/pipeline.py [hunk 1]
```diff
@@ -8,14 +8,34 @@
...
+        condition_name=condition.condition_name, conditions=condition.conditions, is_active=condition.is_active, pipeline_id=condition.pipeline)
+    new_condition.save(db)
+    return new_condition
+
+
+@router.delete("/condition/{condition_id}", response_model=schemas.PipelineCondition)
+def delete_condition(condition_id: int, db: Session = Depends(session)):
+    if condition := db.query(PipelineCondition).get(condition_id):
+        condition.delete(db)
+    return condition
```
Comment (awtkns) [inline backend/api/routes/pipeline.py:36]: Here we should be checking to see if the user has permissions to edit these conditions

File: backend/api/models/pipeline.py [hunk 2]
```diff
@@ -33,6 +34,13 @@ def to_graph(self) -> DiGraph:
         return graph
 
 
+class PipelineCondition(Base):
+    condition_name = Column(String)
+    conditions = Column(JSON)
+    is_active = Column(Boolean)
```
Comment (awtkns) [inline backend/api/models/pipeline.py:40]: What does is_active mean in this context?

File: frontend/store/conditions.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { generic_get, generic_post, generic_delete } from '~/api'
+
```
Comment (awtkns) [inline frontend/store/conditions.js:2]:
```suggestion
const URL = '/pipeline/condition'

```

File: frontend/store/conditions.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      await generic_delete(this, URL)
+      commit('deleteCondition', id)
+      this.$toaster.toastSuccess('Condition deleted!')
+    } catch (err) {
+      this.$toaster.toastError('Could not delete condition')
+    }
+  },
+  async addCondition({ commit }, payload) {
+    try {
+      const URL = '/pipeline/condition'
```
Comment (awtkns) [inline frontend/store/conditions.js:39]:
```suggestion

```

File: frontend/store/conditions.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      commit('setConditions', res)
+      return res
+    } catch (err) {
+      console.log(err)
+    }
+  },
+  async deleteCondition({ commit }, id) {
+    try {
+      const URL = `/pipeline/condition/${id}`
+      await generic_delete(this, URL)
```
Comment (awtkns) [inline frontend/store/conditions.js:30]:
```suggestion
      await generic_delete(this, `${URL}/${id}`)
```

File: frontend/store/conditions.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  deleteCondition: (state, id) => {
+    const index = state.conditions.findIndex(condition => condition.id === id)
+    state.conditions.splice(index, 1)
+  }
+}
+
+export const actions = {
+  async fetchConditions({ commit }) {
+    try {
+      const URL = '/pipeline/condition'
```
Comment (awtkns) [inline frontend/store/conditions.js:19]:
```suggestion
```

**Review by awtkns — CHANGES_REQUESTED** [review]

**Comment by awtkns** [comment]
> also was thinking of putting it in the pipeline builder page but not sure if that would clutter it up too much with too many buttons - so felt like having it in users settings is a nice central place to put it - but idk if maybe it should be in the DicomRuns page or elsewhere perhaps

We can talk an figure out a good place to put it

File: backend/api/models/pipeline.py (hunk 2 above)
Comment (KlamChowder1) [reply backend/api/models/pipeline.py:40]: wasn't sure if we wanted to allow users to be able to turn on/off their conditional runs so I did a "is_active" flag to indicate if we should be checking for these conditions or not sort of thing - we could discuss this some more though

File: backend/api/routes/pipeline.py (hunk 1 above)
Comment (KlamChowder1) [reply backend/api/routes/pipeline.py:36]: maybe we should link a Condition to a User ?

**Comment by KlamChowder1** [comment]
> @KlamChowder1 so what's missing now, just the actual pipelining integration?

yea - just need to figure out how to actually do the checking of conditions to send to a pipeline

File: backend/api/routes/pipeline.py (hunk 1 above)
Comment (awtkns) [reply backend/api/routes/pipeline.py:36]: The fastapi-crud router would almost be useful here haha

**Comment by KlamChowder1** [comment]
added the "subtype" to specify what property of node/patient/study/series we want to wait for before sending to a pipeline 

autopipe-with-subtype

**Comment by awtkns** [comment]
@KlamChowder1 I think we spent today working on the same thing LOL

**Comment by KlamChowder1** [comment]
@awtkns RIP :(

**Comment by awtkns** [comment]
image

**Comment by awtkns** [comment]
form3
 @KlamChowder1

**Comment by awtkns** [comment]
Closed as features are moved to #134

**Pull request closed without merging**